bin/que

```diff
@@ -84,24 +94,51 @@ rescue NameError
   exit 1
 end
 
-queue_name          = options.queue_name     || ENV['QUE_QUEUE'] || Que::Worker::DEFAULT_QUEUE
-wake_interval       = options.wake_interval  || ENV['QUE_WAKE_INTERVAL']&.to_f
-metrics_labels      = options.metrics_labels || {}
+queue_name          = options.queue_name         || ENV['QUE_QUEUE'] || Que::Worker::DEFAULT_QUEUE
+wake_interval       = options.wake_interval      || ENV['QUE_WAKE_INTERVAL']&.to_f
+metrics_labels      = options.metrics_labels     || {}
 priority_threshold  = options.priority_threshold || ENV['QUE_PRIORITY_THRESHOLD'].to_i
-cursor_expiry       = options.cursor_expiry  || 0
+cursor_expiry       = options.cursor_expiry      || 0
+timeout             = options.timeout            || 0
+timeout_message     = options.timeout_message    || ""
 
 Que.logger.info(msg: 'Starting worker', event: 'que.worker.start')
 
 worker = Que::Worker.new(
-  queue: queue_name, wake_interval: wake_interval,
+  queue: queue_name,
+  wake_interval: wake_interval,
   metrics_labels: metrics_labels,
   priority_threshold: priority_threshold,
   lock_cursor_expiry: cursor_expiry
 )
+
 Thread.new { worker.metrics.expose(port: options.metrics_port) } if options.metrics_port
 
-%w[INT TERM].each { |signal| trap(signal) { worker.stop! } }
+stop = false
+%w[INT TERM].each { |signal| trap(signal) { stop = true } }
 
-worker.work_loop
+worker_thread = Thread.new { worker.work_loop }
+
+loop do
+  sleep 0.1
+  break if stop
+end
```

Comment:
iirc you previously used `sleep 0.1 until stop`, which reads more nicely! Easy either way